src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -337,18 +337,17 @@ private bool CyclicDependencyTest(ProtoCore.AssociativeGraph.GraphNode node, ref
                 return true;
             }
 
-            var indexMap = new Dictionary<GraphNode, int>();
-            var lowlinkMap = new Dictionary<GraphNode, int>();
+            var indexList = new int[codeBlock.instrStream.dependencyGraph.GraphList.Count];
```

Comment:
Is this the actual place where we're getting the perf improvement; since you are pre-allocating the size of the array in this case, rather than initializing an empty dictionary that needs to be expanded every other time values are added to it?